### PR TITLE
feat(protocol-designer): add the release notes button to settings

### PR DESCRIPTION
**Pull request by syao1226**
re AUTH-856

# Overview

Add a `View release notes` button to the Settings page, and display the `AnnouncementModal` when the button is clicked.

## Test Plan and Hands on Testing

- Click on the settings icon
- Ensure it matches the [design](https://www.figma.com/design/WbkiUyU8VhtKz0JSuIFA45/Feature%3A-Protocol-Designer-Phase-1?node-id=3150-116483&node-type=canvas&t=VjSlf85g2lJzT7Ed-0)
- Verify that the Announcement modal appears when the 'View release notes' button is clicked.

## Changelog

- Added a button labeled 'View release notes' next to the PD version with a 12px gap
- Create an `AnnouncementModalProp` interface with optional `isViewReleaseNotes` and `onClose` props to control the visibility of the AnnouncementModal, tracking when the user clicks the button and closes the modal
- Used `useLocation` to check if the user is on the Settings page
- Updated tests

### CODE REVIEW

**Comment by koji** [comment]
overall looks good to me but need to modify the layout a little bit.

the space for items (app settings, user settings, and privacy) is 24px
40px is for the space between modal title and the first item, app settings.

border-radius is 8px


Screenshot 2024-09-25 at 4 05 55 PM

Screenshot 2024-09-25 at 4 15 15 PM

File: protocol-designer/src/organisms/AnnouncementModal/index.tsx [hunk 1]
```diff
@@ -16,9 +17,18 @@ import {
...
-export const AnnouncementModal = (): JSX.Element => {
+interface AnnouncementModalProp {
+  isViewReleaseNotes?: boolean
+  onClose?: () => void
+}
+
+export const AnnouncementModal = (
+  props: AnnouncementModalProp
+): JSX.Element => {
+  const { onClose, isViewReleaseNotes } = props
```
Comment (jerader) [inline protocol-designer/src/organisms/AnnouncementModal/index.tsx]: nite 
```suggestion
  const { onClose, isViewReleaseNotes = false } = props
```

File: protocol-designer/src/organisms/AnnouncementModal/index.tsx [hunk 2]
```diff
@@ -27,12 +37,18 @@ export const AnnouncementModal = (): JSX.Element => {
...
 
+  const userClickViewReleaseNotes =
+    location.pathname === '/settings' && isViewReleaseNotes === true
+
   const [
     showAnnouncementModal,
     setShowAnnouncementModal,
-  ] = React.useState<boolean>(userHasNotSeenAnnouncement)
+  ] = React.useState<boolean>(
+    userClickViewReleaseNotes || userHasNotSeenAnnouncement
```
Comment (jerader) [inline protocol-designer/src/organisms/AnnouncementModal/index.tsx]: i think this should be enough logic! then we can remove the useLocation

```suggestion
    userClickViewReleaseNotes || isViewReleaseNotes
```

**Comment by koji** [comment]
Seth merged the following PR so we can use `import { useState } from 'react'`
https://github.com/Opentrons/opentrons/pull/16353

File: protocol-designer/src/organisms/AnnouncementModal/index.tsx [hunk 3]
```diff
@@ -16,7 +16,15 @@ import {
...
 import { useAnnouncements } from './announcements'
 
-export const AnnouncementModal = (): JSX.Element => {
+interface AnnouncementModalProp {
+  isViewReleaseNotes?: boolean
+  onClose?: () => void
+}
+
+export const AnnouncementModal = (
+  props: AnnouncementModalProp
```
Comment (jerader) [inline protocol-designer/src/organisms/AnnouncementModal/index.tsx]: nit: there are multiple props

```suggestion
  props: AnnouncementModalProps
```

**Review by jerader — APPROVED** [review]
nick work! lgtm!

**Pull request merged**